app/Controllers/Validation/ValidateAddressController.php

```diff
@@ -24,11 +24,16 @@ class ValidateAddressController {
 	public function index( Request $request, FunFunFactory $ffFactory ): Response {
 		$this->addressValidator = $ffFactory->newAddressValidator();
 
+		$isCompanyWithContact = $this->handleCompanyWithContact( $request );
+
 		$addressType = $this->getAddressType( $request );
 		if ( $addressType === AddressType::PERSON ) {
 			$nameViolations = $this->getPersonViolations( $request );
 		} elseif ( $addressType === AddressType::COMPANY ) {
 			$nameViolations = $this->getCompanyViolations( $request );
+			if ( $isCompanyWithContact ) {
+				$nameViolations = array_merge( $nameViolations, $this->getPersonViolations( $request ) );
+			}
```

Comment:
This check should go into `getCompanyViolations` to avoid 2 levels of nesting here